test-apps/ui-test-app/src/frontend/appui/contentviews/ScheduleAnimationViewport.tsx

```diff
@@ -99,13 +99,13 @@ class ScheduleAnimationViewport extends React.Component<ScheduleAnimationViewpor
 
     timelineDataProvider = new ScheduleAnimationTimelineDataProvider(viewState);
     if (timelineDataProvider.supportsTimelineAnimation) {
-      if (timelineDataProvider.loadTimelineData()) // eslint-disable-line @typescript-eslint/no-misused-promises
-        return timelineDataProvider;
+      timelineDataProvider.loadTimelineData(); // eslint-disable-line @typescript-eslint/no-floating-promises
+      return timelineDataProvider;
     } else {
       timelineDataProvider = new AnalysisAnimationTimelineDataProvider(viewState);
       if (timelineDataProvider.supportsTimelineAnimation) {
-        if (timelineDataProvider.loadTimelineData()) // eslint-disable-line @typescript-eslint/no-misused-promises
-          return timelineDataProvider;
+        timelineDataProvider.loadTimelineData(); // eslint-disable-line @typescript-eslint/no-floating-promises
+        return timelineDataProvider;

```

Comment:
@DanEastBentley @bsteinbk TypeScript complained in both of these if statements:
>This condition will always return true since this 'Promise<boolean>' is always defined.  **ts(2801)**

Are we really not supposed to be awaiting these calls?

Reply:
It appears that _getTimelineDataProvider should be async and we should await.  Do you want me to pull branch and fix?

Reply:
Just pushed that fix (https://github.com/imodeljs/imodeljs/pull/2326/commits/23035862ce76c9bc0b16a62790bd106d2b975166) - let me know if that looks OK to you.

Reply:
lgtm - thanks!